src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -436,21 +439,19 @@ class MergeTreeData : public IStorage, public WithMutableContext
     /// Returns absolutely all parts (and snapshot of their states)
     DataPartsVector getAllDataPartsVector(DataPartStateVector * out_states = nullptr, bool require_projection_parts = false) const;
 
-    /// Returns all detached parts
-    DetachedPartsInfo getDetachedParts() const;
-
-    static void validateDetachedPartName(const String & name);
+    /// Returns parts in Active state
+    DataParts getDataPartsForInternalUsage() const;
```

Comment:
Because it always returns active data parts and it can be used mostly for internal operations. `getVisibleDataPartsVector` should be used if it's needed for some query.